src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc

```diff
@@ -0,0 +1,188 @@
+  EXPECT_EQ(mobile_apis::Result::APPLICATION_NOT_REGISTERED,
+            static_cast<mobile_apis::Result::eType>(
+                (*result_msg)[am::strings::msg_params][am::strings::result_code]
+                    .asInt()));
+}
+
+TEST_F(AddSubMenuRequestTest, Run_InvalidSubMenuId_UNSUCCESS) {
+  CommandPtr command(CreateCommand<AddSubMenuRequest>());
+
+  MockAppPtr app(CreateMockApp());
+  MessageSharedPtr dummy_sub_menu(CreateMessage(smart_objects::SmartType_Null));
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(app));
+  ON_CALL(*app, FindSubMenu(_)).WillByDefault(Return(dummy_sub_menu.get()));
+
+  MessageSharedPtr result_msg(CatchMobileCommandResult(CallRun(*command)));
+  EXPECT_EQ(mobile_apis::Result::INVALID_ID,
+            static_cast<mobile_apis::Result::eType>(
+                (*result_msg)[am::strings::msg_params][am::strings::result_code]
+                    .asInt()));
+}
+
+TEST_F(AddSubMenuRequestTest, Run_DuplicatedSubMenuName_UNSUCCESS) {
+  CommandPtr command(CreateCommand<AddSubMenuRequest>());
+
+  MockAppPtr app(CreateMockApp());
+
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(app));
+  ON_CALL(*app, FindSubMenu(_))
+      .WillByDefault(Return(static_cast<SmartObject*>(NULL)));
+  ON_CALL(*app, IsSubMenuNameAlreadyExist(_)).WillByDefault(Return(true));
+
+  MessageSharedPtr result_msg(CatchMobileCommandResult(CallRun(*command)));
+  EXPECT_EQ(mobile_apis::Result::DUPLICATE_NAME,
+            static_cast<mobile_apis::Result::eType>(
+                (*result_msg)[am::strings::msg_params][am::strings::result_code]
+                    .asInt()));
+}
+
+TEST_F(AddSubMenuRequestTest, Run_NotValidSubMenuName_UNSUCCESS) {
+  MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
+  // Not valid sub-menu name.
+  (*command_msg)[am::strings::msg_params][am::strings::menu_name] = "\t\n";
+
+  CommandPtr command(CreateCommand<AddSubMenuRequest>(command_msg));
+
+  MockAppPtr app(CreateMockApp());
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(app));
+  ON_CALL(*app, FindSubMenu(_))
+      .WillByDefault(Return(static_cast<SmartObject*>(NULL)));
+
+  MessageSharedPtr result_msg(CatchMobileCommandResult(CallRun(*command)));
+  EXPECT_EQ(mobile_apis::Result::INVALID_DATA,
+            static_cast<mobile_apis::Result::eType>(
+                (*result_msg)[am::strings::msg_params][am::strings::result_code]
+                    .asInt()));
+}
+
+TEST_F(AddSubMenuRequestTest, Run_SUCCESS) {
+  MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
+  (*command_msg)[am::strings::msg_params][am::strings::menu_name] =
+      "valid_sub_menu_name";
+  (*command_msg)[am::strings::msg_params][am::strings::position] =
+      "test_position";
+
+  CommandPtr command(CreateCommand<AddSubMenuRequest>(command_msg));
+
+  MockAppPtr app(CreateMockApp());
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(app));
+  ON_CALL(*app, FindSubMenu(_))
```

Comment:
@LevchenkoS I think tests should test that AddSubmenu is trying ti find correct sub menu in correct application